include/soci/row.h

```diff
@@ -51,6 +53,9 @@ class SOCI_DECL row
     indicator get_indicator(std::size_t pos) const;
     indicator get_indicator(std::string const& name) const;
 
+    void set_session(session *s) { session_ = s; };
```

Comment:
Having both a ctor taking `session*` and the default ctor and a setter for session seems strange, could we avoid having either the first or the last one of them?

Reply:
I think it is possible. I will check.